src/wire/ip.rs

```diff
@@ -126,6 +127,55 @@ impl fmt::Display for Address {
     }
 }
 
+/// A CIDR containing a prefix of an IP subnet.
+pub enum Cidr {
+    Ipv4(Ipv4Cidr),
+}
+
+impl Cidr {
+    /// Construct a new CIDR
+    pub fn new(addr: Address, mask_len: u8) -> Cidr {
+        match addr {
+            Address::Ipv4(addr) => Cidr::Ipv4(Ipv4Cidr::new(addr, mask_len)),
+            Address::Unspecified => {
+                panic!("Can't convert an unspecified address to a CIDR")
+            }
+        }
+    }
+
+    /// Check whether the subnet described by the CIDR conains the address.
+    pub fn contains_ip(&self, addr: Address) -> bool {
+        match (self, addr) {
+            (&Cidr::Ipv4(ref ipv4_cidr), Address::Ipv4(ipv4_addr)) => {
+                ipv4_cidr.contains_ip(ipv4_addr)
+            },
+            _ => false,
+        }
+    }
+
+    /// Return the mask_len of the CIDR.
```

Comment:
Prefix length, not mask length; the mask is as long as the IP address, but has only as many zeroes as the prefix length.